src/utils/format.ts

```diff
@@ -0,0 +1,38 @@
+// ref: https://stackoverflow.com/questions/10420352/converting-file-size-in-bytes-to-human-readable-string
+/**
+ * Format bytes as human-readable text.
+ *
+ * @param {number} bytes Number of bytes.
+ * @param {boolean} si True to use metric (SI) units, aka powers of 1000. False to use
+ *           binary (IEC), aka powers of 1024.
+ * @param {number} dp Number of decimal places to display.
+ *
+ * @return {string} Formatted string.
+ */
+export const humanFileSize = (
+  bytes: number,
+  si: boolean = false,
+  dp: number = 1,
+) => {
+  const thresh = si ? 1000 : 1024;
+
+  if (Math.abs(bytes) < thresh) {
+    return bytes + ' B';
+  }
+
+  const units = si
+    ? ['kB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB']
+    : ['KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB'];
+  let u = -1;
+  const r = 10 ** dp;
+
+  do {
+    bytes /= thresh;
+    ++u;
```

Comment:
Interesting, i don't see pre-incrementing that often but here it does not matter i guess 😊